.github/workflows/pages.yml

```diff
@@ -0,0 +1,44 @@
+---
+name: Deploy GH Pages
+
+on:
+  workflow_dispatch:
+
+jobs:
+  deploy-pages:
+    runs-on: ubuntu-latest
+    name: Build site and deploy to GHPages
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Install node 16
+        uses: actions/setup-node@v3
+        with:
+          node-version: '16'
+          cache: npm
+
+      - name: Install project modules
+        run: npm ci
+
+      - name: Build site
+        run: npm run build
+
+      - name: Configure git
+        run: |
+          git config user.name "${{ github.actor }}"
+          git config user.email "${{ github.actor }}@users.noreply.github.com"
+
+      - name: Checkout pages branch
+        run: git checkout gh-pages
+
+      - name: Copy new files
+        run: cp -rf _site/* .
+
+      - name: Push generated site
+        run: |
+          git add .
```

Comment:
I wonder if adding only `_site` here is better, so we don't deploy the src files

Reply:
We don't need `_site`, this folder is actually _gitignored_.
The `_site` folder is where the site-generator generates our site into.

In the step above this one, I copy the content of the `_site` folder to the `root`,
and in the step above that, I check out the `gh-pages` branch.

The `gh-pages` branch hosts the site files at `root` which also contains the `LICENSE` and the `.gitignore` files,
so the sources are not actually present at the time of the `git add .` command.

Unfortunately, as the `_site` contents can vary with time, I think that anything other than `.git add .` would require high-maintenance.